qiskit/algorithms/minimum_eigen_solvers/adaptvqe2.py

```diff
@@ -0,0 +1,353 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A ground state calculation employing the AdaptVQE algorithm."""
+#figure out which ansatz is used by qiskit nature
+from typing import Optional, List, Tuple, Union, Callable
+
+import copy
+import re
+import logging
+
+import numpy as np
+from qiskit import QiskitError
+
+from qiskit.algorithms import VQE
+from qiskit.algorithms.minimum_eigen_solvers.vqe import VQEResult
+from qiskit.circuit import QuantumCircuit
+from qiskit.opflow import OperatorBase, PauliSumOp, ExpectationBase, CircuitSampler
+from qiskit.opflow.gradients import GradientBase, Gradient
+from qiskit.algorithms.minimum_eigen_solvers import MinimumEigensolver
+from qiskit.circuit.library import EvolvedOperatorAnsatz, RealAmplitudes,PauliEvolutionGate
+from qiskit.circuit.library import NLocal
+from qiskit.utils.validation import validate_min
+from qiskit_nature import ListOrDictType
+from qiskit_nature.algorithms.ground_state_solvers.minimum_eigensolver_factories.minimum_eigensolver_factory import MinimumEigensolverFactory
+from qiskit_nature.exceptions import QiskitNatureError
+from qiskit_nature.circuit.library import UCC
+from qiskit_nature.operators.second_quantization import SecondQuantizedOp
+from qiskit_nature.converters.second_quantization import QubitConverter
+from qiskit_nature.converters.second_quantization.utils import ListOrDict
+from qiskit_nature.problems.second_quantization import BaseProblem
+from qiskit_nature.results import ElectronicStructureResult
+from qiskit_nature.deprecation import deprecate_arguments
+from qiskit.algorithms.optimizers import Optimizer
+from qiskit.utils import QuantumInstance
+from qiskit.providers import BaseBackend
```

Comment:
BaseBackend, a legacy provider interface, is being removed by #7886, and can be removed (import and its uses etc) from this module too.